diff -Naur libwebp-0.3.0.orig/src/dec/vp8.c libwebp-0.3.0/src/dec/vp8.c
--- libwebp-0.3.0.orig/src/dec/vp8.c	2014-10-13 17:06:58.000000000 +0200
+++ libwebp-0.3.0/src/dec/vp8.c	2014-10-13 17:06:18.000000000 +0200
@@ -539,8 +539,7 @@
   {{0, 0, 1, 1}},  {{1, 0, 1, 1}},  {{0, 1, 1, 1}},  {{1, 1, 1, 1}} };
 
 // Macro to pack four LSB of four bytes into four bits.
-#if defined(__PPC__) || defined(_M_PPC) || defined(_ARCH_PPC) || \
-    (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+#if (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
 #define PACK_CST 0x08040201U
 #else
 #define PACK_CST 0x01020408U
